test/mixin/siren-fetch-mixin-lit-test.js

```diff
@@ -0,0 +1,179 @@
+/* global suite, test, fixture, expect, setup, teardown, sinon */
+
+'use strict';
+
+suite('siren-fetch-mixin-lit', function() {
+	var element, sandbox;
+
+	var testAction = { // example from https://github.com/kevinswiber/siren
+		'name': 'add-item',
+		'title': 'Add Item',
+		'method': 'POST',
+		'href': 'http://api.x.io/orders/42/items',
+		'type': 'application/x-www-form-urlencoded',
+		'fields': [
+			{ 'name': 'orderNumber', 'type': 'hidden', 'value': '42' },
+			{ 'name': 'productCode', 'type': 'text' },
+			{ 'name': 'quantity', 'type': 'number' }
+		]
+	};
+
+	var testAction2 = {
+		'name': 'delete-item',
+		'title': 'Delete Item',
+		'method': 'DELETE',
+		'href': 'http://api.x.io/orders/42/items/5',
+		'type': 'application/x-www-form-urlencoded'
+	};
+
+	var testEntity = { // example from https://github.com/kevinswiber/siren
+		'class': ['order'],
+		'properties': {
+			'orderNumber': 42,
+			'itemCount': 3,
+			'status': 'pending'
+		},
+		'entities': [
+			{
+				'class': ['items', 'collection'],
+				'rel': ['http://x.io/rels/order-items'],
+				'href': 'http://api.x.io/orders/42/items'
+			},
+			{
+				'class': ['info', 'customer'],
+				'rel': ['http://x.io/rels/customer'],
+				'properties': {
+					'customerId': 'pj123',
+					'name': 'Peter Joseph'
+				},
+				'links': [
+					{ 'rel': ['self'], 'href': 'http://api.x.io/customers/pj123' }
+				]
+			}
+		],
+		'actions': [
+			{
+				'name': 'add-item',
+				'title': 'Add Item',
+				'method': 'POST',
+				'href': 'http://api.x.io/orders/42/items',
+				'type': 'application/x-www-form-urlencoded',
+				'fields': [
+					{ 'name': 'orderNumber', 'type': 'hidden', 'value': '42' },
+					{ 'name': 'productCode', 'type': 'text' },
+					{ 'name': 'quantity', 'type': 'number' }
+				]
+			}
+		],
+		'links': [
+			{ 'rel': ['self'], 'href': 'http://api.x.io/orders/42' },
+			{ 'rel': ['previous'], 'href': 'http://api.x.io/orders/41' },
+			{ 'rel': ['next'], 'href': 'http://api.x.io/orders/43' }
+		]
+	};
+
+	setup(function() {
+		sandbox = sinon.sandbox.create();
+		element = fixture('basic');
+		element.token = 'foozleberries';
+	});
+
+	teardown(function() {
+		sandbox.restore();
+	});
+
+	suite('smoke test', function() {
+		test('can be instantiated', function() {
+			expect(element.tagName.toLowerCase()).to.equal('siren-fetch-mixin-lit-test-component');
+		});
+	});
+
+	suite('send action unit tests', function() {
+		setup(function() {
+			sinon.stub(window, 'fetch');
```

Comment:
Might want to stub https://github.com/Brightspace/d2l-fetch/blob/master/src/d2lfetch.js#L7 instead of the global window fetch.

That way you only trap the things you are expecting rather than everything that calls fetch on the page